feat(token userview): show user view based on token addresses

simboonlong opened the pull request:
- Holder view
  - show `Transfer Holdership` field.
  - show `Change Beneficiary` field (when there is an beneficiary endorsement awaiting)
- Beneficiary view
  - show `Approve Change Beneficiary` field.
- Holder && Beneficiary view
  - show `Transfer Holdership` field.
  - show `Change Beneficiary` field.
  - show `Surrender Document` field.
- Show a spinner loader when application is fetching token response.
- Show a default no access message when token registry does not match with user token.
- Refactored components.

simboonlong committed:
refactor: get token addresses from reducers to determine which user view to render

simboonlong committed:
refactor: avoid duplicate code in TokenSideBarContent
rename import styles to css for consistency

simboonlong committed:
refactor: get UserRole by switch
test: check if sidebar content renders the correct view

sumitnitsurat commented on src/components/AssetInfo/TokenSideBar.test.tsx:
```diff
@@ -4,7 +4,15 @@ import TokenSideBar from "./TokenSideBar";
+        approvedBeneficiaryAddress=""
+        handler={() => {}}
+      />
+    );
```
write more tests to cover different conditions.
1. if holder === bene
2. approved bene address is there. 
3. if admin address === holder 
4. admin address == bene
5. if adminAddr !== bene and adminAddr !== holder

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarBeneficiary.tsx:
```diff
@@ -0,0 +1,13 @@
+
+const TokenSideBarBeneficiary = () => (
+  <TokenSideBarField title="Allow Transfer" ctaText="Allow" ctaStatus="success">
+    <label>
```
Make the title to "Approve Change Beneficiary". and submit button text to "Approve".

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarContent.test.tsx:
```diff
@@ -1,20 +1,25 @@
-  it("should have 2 fields for Holder with no endorsement awaiting", () => {
-    const wrapper = mount(<TokenSideBarContent isHolder={true} />);
-    expect(wrapper.find("section")).toHaveLength(2);
+  it("should render <TokenSideBarHolder /> view", () => {
```
remove <> from component. can just write "should render TokenSideBarContent" component.

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarContent.test.tsx:
```diff
@@ -1,20 +1,25 @@
-  it("should have 1 field for Beneficiary", () => {
-    const wrapper = mount(<TokenSideBarContent isHolder={false} />);
-    expect(wrapper.find("section")).toHaveLength(1);
+  it("should render <TokenSideBarBeneficiary /> view", () => {
```
same as above.

sumitnitsurat commented on src/components/AssetInfo/TokenSideBarContent.test.tsx:
```diff
@@ -1,20 +1,25 @@
+    );
+    expect(
+      wrapper.containsMatchingElement(
+        <TokenSideBarHolder isEqualBeneficiaryAndHolder={false} approvedBeneficiaryAddress="" />
```
Better to match the content rendered. if just want to test if component is there check for length. rest check what has been rendered from the component.

simboonlong committed:
fix: tests
fix: only show Surrender Document when adminAddr === beneAddr && where adminAddr === holderAddr
fix: only show Approve Change Beneficiary when there is endorsement and in Holder view

simboonlong committed:
feat: show no address match view
fix: text change from transfer ownership > transfer holdership

sumitnitsurat commented on src/components/AssetInfo/TokenSideBar.test.tsx:
```diff
@@ -7,12 +7,147 @@ describe("tokenSideBar", () => {
+          expect(node.find("h4").text()).toStrictEqual("Surrender Document");
+          break;
+        default:
+      }
```
can give the ownership and surrender and other sections id and use that. later on if you change the order test will fail.

simboonlong committed:
fix: incorrect condition of holder + bene without pending endorsement
test: more specific tests on TokenSideBar
feat: show a spinner loader when application is fetching token response.
feat: show a default no access message when token registry does not match with user token.

simboonlong committed:
test(token sidebar reducer): tests on responses when requesting user token (#51)

* test(token sidebar reducer): tests on responses when requesting user token

* test: handle get token addresses saga test cases
